docs/proposals/002-api-proposal/proposal.md

```diff
@@ -80,185 +80,167 @@ The API design is based on these axioms:
-// deployment is created. LLMServerPools require at minimum a single LLMService to
+// deployment is created. InferencePools require at minimum a single InferenceModel to
 // be subscribed to them to accept traffic, any traffic with a model not
-// definied within a LLMService will be rejected.
-type LLMServerPool struct {
+// definied within a InferenceModel will be rejected.
+type InferencePool struct {
         metav1.ObjectMeta
         metav1.TypeMeta
 
-        Spec LLMServerPoolSpec
+        Spec InferencePoolSpec
 }
 
-type LLMServerPoolSpec struct {
+type InferencePoolSpec struct {
         // ModelServerSelector uses label selection to watch model server pods
-        // that should be included in the LLMServerPool. ModelServers should not
-        // be with any other Service or LLMServerPool, that behavior is not supported
+        // that should be included in the InferencePool. ModelServers should not
+        // be with any other Service or InferencePool, that behavior is not supported
         // and will result in sub-optimal utilization.
         ModelServerSelector map[string]string `json:"modelServerSelector,omitempty"`
 }
 ```
 
 ### Yaml Examples
 
-#### LLMServerPool(s)
+#### InferencePool(s)
 Here we create 2 LSPs that subscribe to services to collect the appropriate pods
 ```yaml
 apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMServerPool
+kind: InferencePool
 metadata:
   name: llama-2-pool
   services: 
   - llama-2-vllm
 ---
 apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMServerPool
+kind: InferencePool
 metadata:
   name: gemini-pool
   services: 
   - gemini-jetstream-tpu-v5e
   - gemini-vllm-a100
 ```
 
-#### LLMService
+#### InferenceModel
 
-Here we consume both pools with a single LLMService, while also specifying 2 LLMServices. Where `sql-code-assist` is both the name of the ModelLLMService, and the name of the LoRA adapter on the model server. And `npc-bot` has a layer of indirection for those names, as well as a specified objective. Both `sql-code-assist` and `npc-bot` have available LoRA adapters on both LLMServerPools and routing to each LLMServerPool happens earlier(at the K8s Gateway). So traffic splitting between separate pools happens at the K8s Gateway.
+Here we consume both pools with a single InferenceModel, while also specifying 2 InferenceModels. Where `sql-code-assist` is both the name of the ModelInferenceModel, and the name of the LoRA adapter on the model server. And `npc-bot` has a layer of indirection for those names, as well as a specified objective. Both `sql-code-assist` and `npc-bot` have available LoRA adapters on both InferencePools and routing to each InferencePool happens earlier(at the K8s Gateway). So traffic splitting between separate pools happens at the K8s Gateway.
 ```yaml
 apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMService
+kind: InferenceModel
 metadata:
   name: my-llm-service
 spec:
-  LLMServices:
+  InferenceModels:
```

Comment:
This seems outdated - I think the spec you have above does not use a list here.

Reply:
updated